float64[8] X
float64[8] Y
float64[8] Z
float64[8] K
float64[8] M
float64[8] N

